code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -913,30 +909,24 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	..()
 
 // Here so admins can unemag borgs.
-/mob/living/silicon/robot/proc/unemag()
-	emagged = FALSE
+/mob/living/silicon/robot/unemag()
+	SetEmagged(FALSE)
 	if(!module)
 		return
 
-	var/list/all_modules = contents | module.modules
-	for(var/obj/item/I in all_modules)
-		if(!(I in module.emag_modules)) // If `I` is not an emagged module, skip it.
-			continue
-
-		// Clear the emagged item from any of their active modules.
+	// Deletes the emag items, and recreates them.
+	// This isn't that clean but it prevents shenanigans where emag items would hang around in the inventory.
+	for(var/obj/item/I in module.emag_modules)
```

Comment:
this list can only contain items right? If so it can be made typeless

Reply:
Shouldn't this also be part of the `module/unemag` proc?

Reply:
Yes it can be made typless, and yeah I'll put it into `module/unemag` instead. Also, this made me take a look at this proc and realize I'm missing some logic, which I will be adding in.